utils/common/src/main/kotlin/DirectoryStash.kt

```diff
@@ -32,24 +32,34 @@ import kotlin.io.path.moveTo
 import org.apache.logging.log4j.kotlin.Logging
 
 /**
- * A convenience function that stashes directories using a [DirectoryStash] instance.
+ * A convenience function that stashes [directories] to the [stashParentDir] using a [DirectoryStash] instance. If
+ * [deleteParentOnClose] is true, the [stashParentDir] is expected to be empty on close and will be deleted.
  */
-fun stashDirectories(vararg directories: File): Closeable = DirectoryStash(setOf(*directories))
+fun stashDirectories(
+    vararg directories: File,
+    stashParentDir: File = createTempDirectory("stashParent").toFile(),
```

Comment:
commit-message: Please provide a rationale _why_ the parameters are introduced.

Reply:
This became irrelevant due to the new approach.